src/components/shared/page.tsx

```diff
@@ -0,0 +1,113 @@
+// ----- Imports ----- //
+
+import React from 'react';
+import { css } from '@emotion/core';
+
+import Article from 'components/news/article';
+import LiveblogArticle from 'components/liveblog/liveblogArticle';
+import OpinionArticle from 'components/opinion/opinionArticle';
+import { Pillar, pillarFromString } from 'pillar';
+import { Content } from 'capiThriftModels';
+import { includesTweets } from 'capi';
+import { fontFace } from 'styles';
+import { None, Some } from 'types/option';
+
+
+// ----- Components ----- //
+
+const PageStyles = css`
+    ${fontFace("Guardian Egyptian Web", new None, "/public/fonts/egyptiantext-regular.otf")}
+    ${fontFace("Guardian Egyptian Web", new Some(500), "/public/fonts/egyptiantext-medium.otf")}
+    ${fontFace("Guardian Egyptian Web", new Some("bold"), "/public/fonts/egyptiantext-medium.otf")}
+    ${fontFace("Guardian Text Sans Web", new None, "/public/fonts/GuardianSansWeb-Regular.ttf")}
+    ${fontFace("Guardian Headline", new Some(200), "/public/fonts/GHGuardianHeadline-Light.ttf")}
+    ${fontFace("Guardian Headline", new Some(400), "/public/fonts/GHGuardianHeadline-Regular.ttf")}
+    ${fontFace("Guardian Headline", new Some(500), "/public/fonts/GHGuardianHeadline-Medium.ttf")}
+    ${fontFace("Guardian Headline", new Some(700), "/public/fonts/GHGuardianHeadline-Bold.ttf")}
+    ${fontFace("Guardian Icons", new None, "/public/fonts/icons.otf")}
+
+    font-size: 62.5%;
+    background: white;
+
+    body {
+        margin: 0;
+        font-family: 'Guardian Egyptian Web';
+        font-size: 1.6em;
+        line-height: 1.5em;
+        overflow-x: hidden;
+    }
+
+    figure {
+        margin: 1em 0;
+    }
+
+    figure.element-embed {
+        margin: 2em 0
+    }
+
+    .js-email-sub__iframe {
+        width: 100%;
+    }
+
+    .js-email-sub__iframe + figcaption {
+        margin-top: -8px;
+    }
+
+    video,
+    .element-atom iframe,
+    .element-audio iframe {
+        width: 100%;
+    }
+`;
+
+function ArticleBody(props: { imageSalt: string, capi: Content }): React.ReactElement {
```

Comment:
Ah was literally rewriting this myself :)

Reply:
Turns out there's a few different types of articles with a capi type of 'article', including `immersives`

Reply:
Yeah I think we may have to write a data type for our own definition of article "type". Then we can map the MAPI response onto it:

```ts
const enum Article {
  STANDARD,
  OPINION,
  IMMERSIVE,
  LIVEBLOG,
}

const parseArticle(content: Content): Article {
  ...
}
```